File: AngularJS/src/org/angular2/entities/Angular2DirectiveAttribute.java
package org.angular2.entities;

import com.intellij.lang.javascript.psi.JSType;
import com.intellij.model.Pointer;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiElement;
import org.angular2.web.Angular2PsiSourcedSymbol;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import static org.angular2.web.Angular2WebSymbolsAdditionalContextProvider.KIND_NG_DIRECTIVE_ATTRIBUTES;

public interface Angular2DirectiveAttribute extends Angular2PsiSourcedSymbol, Angular2Element {

  @Override
  @NotNull
  String getName();

  @Nullable
  @Override
  JSType getType();

  @NotNull
  @Override
  Pointer<? extends Angular2DirectiveAttribute> createPointer();

  @NotNull
  @Override
  default PsiElement getSource() {
    return getSourceElement();
  }

  @Override
  default @NotNull Project getProject() {
    return getSourceElement().getProject();
  }

  @NotNull
  @Override
  default String getNamespace() {
    return NAMESPACE_JS;
  }

  @NotNull
  @Override
  default String getKind() {
    return KIND_NG_DIRECTIVE_ATTRIBUTES;
  }
}

File: AngularJS/src/org/angular2/entities/Angular2DirectiveAttribute.kt
// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
package org.angular2.entities

import com.intellij.lang.javascript.psi.JSType
import com.intellij.model.Pointer
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiElement
import com.intellij.webSymbols.WebSymbol
import org.angular2.web.Angular2PsiSourcedSymbol

import org.angular2.web.Angular2WebSymbolsQueryConfigurator.Companion.KIND_NG_DIRECTIVE_ATTRIBUTES

interface Angular2DirectiveAttribute : Angular2PsiSourcedSymbol, Angular2Element {

  override val name: String

  override val type: JSType?

  override val source: PsiElement
    get() = sourceElement

  override val project: Project
    get() = sourceElement.project

  override val namespace: String
    get() = WebSymbol.NAMESPACE_JS

  override val kind: String
    get() = KIND_NG_DIRECTIVE_ATTRIBUTES

  override val apiStatus: WebSymbol.ApiStatus?

  override fun createPointer(): Pointer<out Angular2DirectiveAttribute>
}
